src/openqaoa-core/problems/portfoliooptimization.py

```diff
@@ -0,0 +1,223 @@
+            for j in range(num_assets):
+                sigma[i][j] = (
+                    sigma_bounds[1] - sigma_bounds[0]
+                ) * np.random.rand() + sigma_bounds[0]
+
+        return PortfolioOptimization(mu, sigma, risk_factor, budget, penalty)
+
+    @property
+    def docplex_model(self):
+        mdl = Model("portfolio_optimization")
+
+        num_assets = self.num_assets
+
+        # Start the docplex model with Model("name of the model")
+        mdl = Model("Portfolio Optimization")
+
+        # Consider the number of variables as num_assets,
+        # and binary set of variables that represent the stocks
+        # x vector in numpy array for matrix multiplication
+        x = np.array(mdl.binary_var_list(num_assets, name="asset"))
+
+        # Specific the objective of the
+        # portfolio optimization function
+        objective_function = -np.array(self.mu) @ x + x.T @ np.array(self.sigma) @ x
+
+        # For this problem it aims to maximize the profit
+        # of those assets minimizing the risk of the investment
+        mdl.minimize(objective_function)
+
+        # Budget constraint
+        mdl.add_constraint(mdl.sum(x) == self.budget, ctname="budget")
+
+        return mdl
+
+    @property
+    def qubo(self):
+        """
+        Returns the QUBO encoding of this problem.
+        Returns
+        -------
+            The QUBO encoding of this problem.
+        """
+        cplex_model = self.docplex_model
+        n_vars = cplex_model.number_of_binary_variables
+        qubo_docplex = FromDocplex2IsingModel(cplex_model, multipliers=self.penalty)
+        ising_model = qubo_docplex.ising_model
+        return QUBO(
+            n_vars,
+            ising_model.terms + [[]],
+            ising_model.weights + [ising_model.constant],
+            self.problem_instance,
+        )
+
+    def classical_solution(self, string: bool = False):
+        """
+        Return the classical solution of the bin packing problem
+        Parameters
+        ----------
+        string : bool, optional
+            If the solution is returned as a string. The default is False.
+        Raises
+        ------
+        ValueError
+            A flag if docplex does not find a valid solution.
+        Returns
+        -------
+        solution : Union[str, dict]
+            The classical solution of the specific problem as a string or a dict.
+        """
+        cplex_model = self.docplex_model
+        docplex_sol = cplex_model.solve()
+
+        if docplex_sol is None:
+            raise ValueError(f"solution not found: {cplex_model.solve_details.status}")
+
+        if string:
+            solution = ""
+        else:
+            solution = {}
+        for var in cplex_model.iter_binary_vars():
+            if string:
+                solution += str(int(np.round(docplex_sol.get_value(var), 1)))
+            else:
+                solution[var.name] = int(np.round(docplex_sol.get_value(var), 1))
```

Comment:
round is used because sometimes docplex gives the solution close to 1 but not 1 (in a docplex format), if we want to store this as an integer and use int(value) and the value is 0.999999 it will store a zero. I changed it for just round(docplex_sol.get_value(var)) which gives the same result.